### DISCUSSION TITLE
Worstfish (Stockfish where it selects the worst move)

### DISCUSSION THREAD

**Discussion by TheDrawingCoder-Gamer** [official-stockfish/Stockfish#4645] [General]
I have seen Tom7's (suckerpinch's) video on weird chess algorithims, and one of them is using stockfish but it selects the worst move. He added in the video that it technically isn't the worst move because it assumes that stockfish will be playing at its best. Is there a way for the minimax algorithim to become a minimini algorithim (or a maximax algorithim) by assuming that stockfish will be selecting the worst move?

**Comment by peregrineshahin** [comment]
> I have seen Tom7's (suckerpinch's) video on weird chess algorithims, and one of them is using stockfish but it selects the worst move. He added in the video that it technically isn't the worst move because it assumes that stockfish will be playing at its best. Is there a way for the minimax algorithim to become a minimini algorithim (or a maximax algorithim) by assuming that stockfish will be selecting the worst move?

I watched the part of that video so that I can understand what you/he was referring to. 
The worst move can indeed be obtained by a maximin algorithm instead of what you're trying to find, Pay a little attention to what he meant, it wasn't that he really implemented a maximin algorithm to obtain the worst move but rather he only have chosen the worst move at the root node after the typical Stockfish search (basically can be done by enabling multiPvs and picking the last move from PVs no changes to the algorithm) which indeed wouldn't give you the worst possible move because you are reversing only the last outcome of the good moves search and not really searching for worst moves.
https://youtu.be/DpXy041BIlA?t=1256

**Comment by ChessOverflow** [comment]
Worsfish has a simple formula. Try to move the king to the center. Just do it.

Also, if there is a match between the worstfish and the bestfish, it will be a maximum of 2 moves.

1.f3 e6 2.g4 Qh4#, or
1.f3 e5 2.g4 Qh4#, or
1.f4 e6 2.g4 Qh4#, or
1.f4 e5 2.g4 Qh4#, or
1.g4 e6 2.f3 Qh4#, or
1.g4 e6 2.f4 Qh4#, or
1.g4 e5 2.f3 Qh4#, or
1.g4 e5 2.f4 Qh4#
